target-riscv/cpu.c

```diff
@@ -96,16 +92,7 @@ static void riscv_cpu_disas_set_info(CPUState *s, disassemble_info *info) {
 static void riscv_cpu_realizefn(DeviceState *dev, Error **errp)
 {
     CPUState *cs = CPU(dev);
-    RISCVCPU *cpu = RISCV_CPU(dev);
     RISCVCPUClass *mcc = RISCV_CPU_GET_CLASS(dev);
-    CPURISCVState *env = &cpu->env;
-
-    /* Enable GC ISA */
-    set_feature(env, RISCV_FEATURE_RVM);
```

Comment:
Presumably AArch64 supports something similar in QEMU for the AArch32 mode change? Does it have a more efficient way of handling that?

For what it's worth, my understanding is that the spec doesn't require that writes to `misa` are supported - it could be that attempts to modify `misa` are a no-op.